integrations/lambda/demo.ipynb

```diff
@@ -1423,21 +1423,712 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "from botocore.exceptions import ClientError\n",
+    "import json\n",
+    "import time\n",
+    "import requests"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f10bfbc6",
+   "metadata": {},
+   "source": [
+    "### Create Role for Workspace"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c5fcf4ef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize IAM and Managed Grafana clients\n",
+    "iam_client = boto3.client('iam', region_name=REGION_NAME)\n",
+    "grafana_client = boto3.client('grafana', region_name=REGION_NAME)\n",
+    "\n",
+    "workspace_role_arn = \"\"\n",
+    "\n",
+    "# Define the trust policy for Amazon Managed Grafana\n",
+    "trust_policy = {\n",
+    "    \"Version\": \"2012-10-17\",\n",
+    "    \"Statement\": [\n",
+    "        {\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Principal\": {\n",
+    "                \"Service\": \"grafana.amazonaws.com\"\n",
+    "            },\n",
+    "            \"Action\": \"sts:AssumeRole\"\n",
+    "        }\n",
+    "    ]\n",
+    "}\n",
+    "\n",
+    "workspace_role_name = 'GrafanaWorkspaceRole'\n",
+    "try:\n",
+    "    # Create the IAM role\n",
+    "    create_role_response = iam_client.create_role(\n",
+    "        RoleName=workspace_role_name,\n",
+    "        AssumeRolePolicyDocument=json.dumps(trust_policy),\n",
+    "        Description=\"Role for Amazon Managed Grafana to access AWS resources\"\n",
+    "    )\n",
+    "    workspace_role_arn = create_role_response['Role']['Arn']\n",
+    "except iam_client.exceptions.EntityAlreadyExistsException:\n",
+    "    print(f\"Workspace IAM role {role_name} already exists\")\n",
+    "    try:\n",
+    "        workspace_role_arn = iam_client.get_role(RoleName=workspace_role_name)['Role']['Arn']\n",
+    "    except iam_client.exceptions.NoSuchEntityException:\n",
+    "        print(\"Workspace IAM role could not be found\")\n",
+    "        raise\n",
+    "    \n",
+    "print(f\"Created workspace role with ARN: {workspace_role_arn}\")\n",
+    "\n",
+    "# Define an inline policy for Timestream and CloudWatch read access\n",
+    "inline_policy = {\n",
+    "    \"Version\": \"2012-10-17\",\n",
+    "    \"Statement\": [\n",
+    "        {\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": [\n",
+    "                \"timestream:DescribeEndpoints\",\n",
+    "                \"timestream:ListDatabases\"\n",
+    "            ],\n",
+    "            \"Resource\": \"*\"\n",
+    "        },\n",
+    "        {\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": [\n",
+    "                \"timestream:DescribeTable\",\n",
+    "                \"timestream:Select\",\n",
+    "                \"timestream:SelectValues\",\n",
+    "                \"timestream:ListMeasures\"\n",
```

Comment:
Curious why permissions like `ListMeasures` are needed. My dashboard for InfluxDB only required the following permissions:

```
                  - timestream:DescribeEndpoints
                  - timestream:ListDatabases
                  - timestream:ListTables
                  - timestream:Select
```

Reply:
I have changed the role's permissions to just the ones you listed.